File: common/src/main/java/net/shiroha233/roadweaver/structures/registry/ModStructureTypes.java
package net.shiroha233.roadweaver.structures.registry;

import net.minecraft.world.level.levelgen.structure.StructureType;
import net.shiroha233.roadweaver.structures.types.BridgeTemplateStructure;
import net.shiroha233.roadweaver.structures.types.RoadsideStructure;
import net.shiroha233.roadweaver.structures.types.RoadsideVillageStructure;
import net.shiroha233.roadweaver.structures.types.SpawnCabinStructure;

/**
 * 模组结构类型注册中心
 */
public final class ModStructureTypes {
    private ModStructureTypes() {}
    
    public static StructureType<RoadsideStructure> ROADSIDE;
    public static StructureType<RoadsideVillageStructure> ROADSIDE_VILLAGE;
    public static StructureType<SpawnCabinStructure> SPAWN_CABIN;
    public static StructureType<BridgeTemplateStructure> BRIDGE_TEMPLATE;
    
    public static void setRoadside(StructureType<RoadsideStructure> type) {
        ROADSIDE = type;
    }

    public static void setRoadsideVillage(StructureType<RoadsideVillageStructure> type) {
        ROADSIDE_VILLAGE = type;
    }
    
    public static void setSpawnCabin(StructureType<SpawnCabinStructure> type) {
        SPAWN_CABIN = type;
    }

    public static void setBridge(StructureType<BridgeTemplateStructure> type) {
        BRIDGE_TEMPLATE = type;
    }
}

File: common/src/main/kotlin/net/shiroha233/roadweaver/structures/registry/ModStructureTypes.kt
package net.shiroha233.roadweaver.structures.registry

import net.minecraft.world.level.levelgen.structure.StructureType
import net.shiroha233.roadweaver.structures.types.RoadsideStructure
import net.shiroha233.roadweaver.structures.types.SpawnCabinStructure

/**
 * 模组结构类型注册
 *
 * 由 Fabric/NeoForge 平台各自实现注册逻辑，
 * 这里只定义引用。
 */
@Suppress("MemberVisibilityCanBePrivate")
object ModStructureTypes {
    /**
     * 路边结构类型
     */
    @JvmField
    var ROADSIDE: StructureType<RoadsideStructure>? = null

    /**
     * 初始小屋结构类型
     */
    @JvmField
    var SPAWN_CABIN: StructureType<SpawnCabinStructure>? = null

    /**
     * 设置路边结构类型（由平台注册时调用）
     */
    @JvmStatic
    fun setRoadside(type: StructureType<RoadsideStructure>) {
        ROADSIDE = type
    }

    /**
     * 设置初始小屋结构类型（由平台注册时调用）
     */
    @JvmStatic
    fun setSpawnCabin(type: StructureType<SpawnCabinStructure>) {
        SPAWN_CABIN = type
    }
}
